app/views/ThemeView.tsx

```diff
@@ -58,79 +57,112 @@ interface ITheme {
 	group: string;
 }
 
-type IThemeViewProps = IBaseScreen<SettingsStackParamList, 'ThemeView'>;
+const Item = ({
+	onPress,
+	label,
+	value,
+	isSelected
+}: {
+	onPress: () => void;
+	label: string;
+	value: string;
+	isSelected: boolean;
+}) => {
+	const { colors } = useTheme();
+	return (
+		<>
+			<List.Item
+				title={label}
+				onPress={onPress}
+				testID={`theme-view-${value}`}
+				right={() => (isSelected ? <List.Icon name='check' color={colors.tintColor} /> : null)}
+			/>
+			<List.Separator />
+		</>
+	);
+};
 
-const ThemeView = ({ navigation }: IThemeViewProps): React.ReactElement => {
-	const { theme, themePreferences, setTheme } = useTheme();
+const ThemeView = (): React.ReactElement => {
+	const { themePreferences, setTheme } = useTheme();
+	const { setOptions } = useNavigation();
 
 	useLayoutEffect(() => {
-		navigation.setOptions({
+		setOptions({
 			title: I18n.t('Theme')
 		});
-	}, [navigation]);
-
-	const isSelected = (item: ITheme) => {
-		const { group } = item;
-		const { darkLevel, currentTheme } = themePreferences as IThemePreference;
-		if (group === THEME_GROUP) {
-			return item.value === currentTheme;
-		}
-		if (group === DARK_GROUP) {
-			return item.value === darkLevel;
-		}
-	};
+	}, []);
+
+	const isSelected = useCallback(
+		(item: ITheme) => {
+			const { group } = item;
+			const { darkLevel, currentTheme } = themePreferences as IThemePreference;
+			if (group === THEME_GROUP) {
+				return item.value === currentTheme;
+			}
+			if (group === DARK_GROUP) {
+				return item.value === darkLevel;
+			}
+		},
+		[themePreferences?.currentTheme, themePreferences?.darkLevel]
+	);
 
-	const onClick = (item: ITheme) => {
-		const { darkLevel, currentTheme } = themePreferences as IThemePreference;
-		const { value, group } = item;
-		let changes: Partial<IThemePreference> = {};
-		if (group === THEME_GROUP && currentTheme !== value) {
-			logEvent(events.THEME_SET_THEME_GROUP, { theme_group: value });
-			changes = { currentTheme: value as TThemeMode };
-		}
-		if (group === DARK_GROUP && darkLevel !== value) {
-			logEvent(events.THEME_SET_DARK_LEVEL, { dark_level: value });
-			changes = { darkLevel: value as TDarkLevel };
-		}
-		_setTheme(changes);
-	};
+	const onClick = useCallback(
```

Comment:
Not really doing anything apparently (or almost nothing).
The only state we have is `themePreferences` and you had to add it as dependencies.
It might be less memory consuming to just recreate it on every render.